src/components/chat-nav-item/ChatNavItem.ts

```diff
@@ -1,14 +1,38 @@
 import BaseBlock from '../../common/base-block';
 import './style.less';
 import template from './template.hbs';
+import {Context} from "../../helpers/context";
+import {BaseBlockOptions, BaseBlockProps} from "../../common/types";
+import {ChatData} from "../../data";
+
+type ChatNavItemOptions  = BaseBlockOptions & {
+    chat: ChatData;
+}
 
 class ChatNavItem extends BaseBlock {
+    protected _props: BaseBlockProps & {
+        chat: ChatData,
+    }
```

Comment:
Сейчас с первого взгляда не очень понятно чем отличаются Options и Props и зачем было сделано это разделение. Так же не совсем понятно зачем объединять типы Options и Props дочернего компонента с базовым, они по логике являются разными сущностями.
Я бы предложил сделать класс Block параметризуемым с помощью generic'ов, чтобы можно было писать так:
```suggestion
type ChatNavItemOptions  = BaseBlockOptions & {
    chat: ChatData;
}

class ChatNavItem extends BaseBlock<ChatNavItemProps> {
    protected _props: BaseBlockProps & {
        chat: ChatData,
    }
```
В таком случае по идее дочерний компонент уже не должен будет ничего знать о пропсах базового компонента